### PR TITLE
[web_editor] create oe_structure view in extension mode

### PR DESCRIPTION
**Pull request by hsi-x**
issue occurs due view mode is saved as base view not as extension view.

before commit:

    when user open website on edit mode and drag and drop snippet from
    snippet panel to sidebar and click save button.changes of user is not
    visible in sidebar

    for ex. if user open open website in edit mode and drag and drop
    Title snippet from snippet panel to Follow Us in sidebar and click 
    on save button website reload automatically after reload website Title 
    snippet is not visible in Follow Us in sidebar.

task-2311520

### CODE REVIEW

File: addons/web_editor/models/ir_ui_view.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ def save_oe_structure(self, el):
             'arch': self._pretty_arch(arch),
             'key': '%s_%s' % (self.key, el.get('id')),
             'type': 'qweb',
+            'mode': 'extension'
```
Comment (qsm-odoo) [inline addons/web_editor/models/ir_ui_view.py:84]: Oh... indeed but why does it work for some oe_structure then? :scream:

File: addons/web_editor/tests/test_views.py [hunk 2]
```diff
@@ -17,11 +17,35 @@ def test_infinite_inherit_loop(self):
...
+             data-oe-xpath="/xpath/div" data-oe-model="ir.ui.view" data-oe-field="arch">
+            <div>Please contact to JPR for more details of this product!</div></div>''' % third_view.id
+
+        third_view.save(value=value, xpath='/xpath/div')
+
+        self.assertEqual(len(third_view.inherit_children_ids), 1)
+        self.assertEqual(third_view.inherit_children_ids.mode, 'extension')
+        self.assertIn('<div>Please contact to JPR for more details of this product!</div>',
+            third_view.inherit_children_ids.read_combined(['arch'])['arch'],
+        )
```
Comment (qsm-odoo) [inline addons/web_editor/tests/test_views.py:51]: This test does not seem related to `test_infinite_inherit_loop` you are modifying. If there is no test for oe_structure saving yet, create a new one.

File: addons/web_editor/models/ir_ui_view.py (hunk 1 above)
Comment (qsm-odoo) [reply addons/web_editor/models/ir_ui_view.py:84]: @hsi-odoo In fact I see that in the case where it works, the mode is indeed set to "extension" automatically... do you know why?

Comment (qsm-odoo) [reply addons/web_editor/models/ir_ui_view.py:84]: @hsi-odoo We found the issue, in fact it is because the create is done using `self` two lines below. Could you also change that into `self.env['ir.ui.view'].create` to make it more explicit? Thanks :)

**Comment by nim-odoo** [comment]
note to myself: opw 2334904

**Comment by JKE-be** [comment]
since no new of author of the PR following our review, I fixed it myself.

Rewrite test that fails before fix
Explain double fix in commit

@robodoo r+
